File: src/mc_tut1/src/robot_news_int.cpp
#include "example_interfaces/msg/int32.hpp"
#include "rclcpp/rclcpp.hpp"

class RobotnewInt: public rclcpp::Node{
    public:
    RobotnewInt(): Node("Robot_News_Int"), count(0){
        publisher = this->create_publisher<example_interfaces::msg::Int32>("robot_news_int", 10);
        timer_ = this->create_wall_timer(std::chrono::seconds(2), std::bind(&RobotnewInt::publishnews, this));
        RCLCPP_INFO(this->get_logger(), "Robot news station started");
    }

    void publishnews(){
        auto msg = example_interfaces::msg::Int32();
        msg.data = count;
        publisher->publish(msg);
        count++;
    }

    private:
    rclcpp::TimerBase::SharedPtr timer_;
    rclcpp::Publisher<example_interfaces::msg::Int32>::SharedPtr publisher;
    int count;
};

int main(int argc, char** argv){
    rclcpp::init(argc, argv);
    auto node = std::make_shared<RobotnewInt>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: src/my_package_tut1/my_package_tut1/robot_news_int.py
import rclpy
from rclpy.node import Node
from example_interfaces.msg import Int32
class Mynode(Node):
    def __init__(self):
        super().__init__("robot_station")
        self.declare_parameter("number", 200)
        self.declare_parameter("timer_period", 0.5)
        self.data = self.get_parameter("number").value
        self.timer = self.get_parameter("timer_period").value

        self.robot_name = "Shimbs"
        self.publisher1 = self.create_publisher(Int32,"data",10)
        self.create_timer(self.timer, self.publishe_news)
        self.get_logger().info("Robot news station INT started")



    def publishe_news(self):
        msg = Int32()
        msg.data = self.data
        self.publisher1.publish(msg)


def main(args=None):
    rclpy.init(args=args)
    node = Mynode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == '__main__':
    main()
